pkg/sources/git/git.go

```diff
@@ -722,18 +735,22 @@ func (s *Git) ScanRepo(ctx context.Context, repo *git.Repository, repoPath strin
 		}
 	}
 
+	logger := ctx.Logger()
 	// We're logging time, but the repoPath is usually a dynamically generated folder in /tmp.
 	// To make this duration logging useful, we need to log the remote as well.
-	remotes, _ := repo.Remotes()
-	repoURL := "Could not get remote for repo"
-	if len(remotes) != 0 {
-		repoURL = getSafeRemoteURL(repo, remotes[0].Config().Name)
+	// However, the context may already contain a key for "repo".
```

Comment:
What are the circumstances under which this happens? _Should_ it be happening?

Reply:
The GitHub source passes a context with `repo` because (a) we already know the remoteURL, and (b) including that information in pre-scan logs might be useful. The GitLab source does not include this information but it's arguably useful to have. The Git source is written with the assumption that the remoteURL is attached (if available) in the `ScanRepo` function.

I don't know if I can say whether it _should_ be happening, but having duplicate keys is a bit annoying.